torch/_export/graph_module.py

```diff
@@ -86,6 +185,67 @@ def recompile(self) -> torch.fx.graph_module.PythonCode:
         self._code = python_code.src
         return python_code
 
+    def __reduce__(self) -> Tuple[Callable[..., "ExportGraphModule"], Tuple[bytes]]:
+        """
+        Serialization of the ExportGraphModule. The FX serialization does not
+        serialize the underlying graph to preserve backwards-compatiblity and
+        instead retraces the graph module when loading.  This results in loss of
+        metadata that is later used for optimizations directly on the graph
+        module.  Since we want to preserve this metadata and we do not care that
+        much about BC, we will write our own serialization method.
+        """
+
+        def op_to_str(op):
+            # Replace the targets with their names since we cannot serialize the ops
+            if isinstance(op, torch._ops.HigherOrderOperator):
+                return f"torch.ops.{op.__name__}"
+            elif "torch" in op.__module__:
+                return str(op)
+            else:
+                warnings.warn(f"Could not convert op {op} to str")
+                return ""
+
+        gm_dict = self.__dict__.copy()
+        gm_dict["_graphmodule_cls_name"] = self.__class__.__name__
+
+        graph = copy.deepcopy(gm_dict["_graph"])
+        for node in graph.nodes:
+            # Replace the ops with their names since we cannot serialize the ops
+            if node.op == "call_function":
+                node.target = op_to_str(node.target)
+
+            if (original_aten := node.meta.get("original_aten", None)) is not None:
+                node.meta["original_aten"] = op_to_str(original_aten)
+
+            if (source_fn := node.meta.get("source_fn", None)) is not None:
+                node.meta["source_fn"] = op_to_str(source_fn)
+
+            # Replace the faketensor metadata with the tensor metadata dataclass
+            # since we cannot serialize faketensors
+            if (val := node.meta.get("val", None)) is not None:
+                node.meta["val"] = pytree.tree_map_only(
```

Comment:
node.meta['val'] could be SymInt too

Here's one example of how this could happen

`class DynamicShapeSimpleModel(torch.nn.Module):
    def __init__(self):
        super().__init__()

    def forward(self, a, b, c) -> torch.Tensor:
        d = (torch.matmul(a, b) + c) / 2
        d_s0 = d.shape[0]
        d_s1 = d.shape[1]
        d_s3 = d_s0 * d_s1
        e = d.view(d_s3)
        return torch.cat([e, e])
`

Reply:
Pickle is able to pickle the symints too.